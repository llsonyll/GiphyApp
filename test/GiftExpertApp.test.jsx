import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import GiftExpertApp from "../src/GifExpertApp";

describe("GiftExpertApp Test", () => {
  test("should render GiftExpertApp component", () => {
    const { container } = render(<GiftExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test("should render GiftExpertApp component", async () => {
    render(<GiftExpertApp />);
    const newCategory = "World of Warcraft";
    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");
    fireEvent.input(input, {
      target: {
        value: newCategory,
      },
    });
    // fireEvent.submit(form);
    // console.log(screen.getAllByRole("heading", { level: 1 }));
    // await waitFor(
    //   expect(
    //     screen.getAllByRole("heading", { level: 1 }).length
    //   ).toBeGreaterThan(1)
    // );
    // expect(await screen.findByText(newCategory)).toBeDefined();
  });
});
