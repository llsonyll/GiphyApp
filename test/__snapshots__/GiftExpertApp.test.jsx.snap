// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftExpertApp Test should render GiftExpertApp component 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <div
    class="addCategory"
  >
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
    </form>
  </div>
  <h3>
     
    One Punch
  </h3>
  <p
    class="loading"
  >
     Cargando... 
  </p>
  <ol
    class="card-grid"
  />
</div>
`;
